### PR TITLE
Switch to using s3 for lfs

### PR DESCRIPTION
**Pull request by emostov**

### CODE REVIEW

File: .github/actions/lfs-setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+      with:
+        go-version: '1.21'
+        cache: false
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+      with:
+        aws-region: us-east-1
+        role-to-assume: arn:aws:iam::799078726966:role/github-mono
```
Comment (james-callahan) [inline .github/actions/lfs-setup/action.yml]: This won't work: that role can only ever be assumed by CI running in the mono repo.
This repo (qos) has it's own role `arn:aws:iam::799078726966:role/github-qos` to use.

Comment (emostov) [reply .github/actions/lfs-setup/action.yml]: [40ecdb6](https://github.com/tkhq/qos/pull/397/commits/40ecdb635ee62b8ba6c2d40f999bd80f46d597e2)

Comment (emostov) [reply .github/actions/lfs-setup/action.yml]: Realizing we don't even need this so its irrelevant

Comment (james-callahan) [reply .github/actions/lfs-setup/action.yml]: > Realizing we don't even need this so its irrelevant

How do we pull from S3 then? Doesn't that require logging into AWS?

Comment (emostov) [reply .github/actions/lfs-setup/action.yml]: Updated with a fixed role

File: .github/workflows/artifacts.yml [hunk 2]
```diff
@@ -35,7 +56,7 @@ jobs:
           path: ${{steps.cache-key.outputs.filenames}}
 
       - name: Setup Checkout
-        if: steps.cache.outputs.cache-hit != 'true'
```
Comment (emostov) [inline .github/workflows/artifacts.yml]: TODO: once CI runs green, comment this back in

**Pull request merged**